### ISSUE TITLE
Inconsistent grading of many text entry answers

### ISSUE THREAD

**Issue by conradteper** [marcusgreen/moodle-qtype_gapfill#98]
Issue with many users' answers marked as incorrect when they have the correct answer entered. Other students marked as correct for the same entry.

HTML source code is as follows:
`<p>28. Many students [agree that the college shouldn't raise] (agree / college / raise / shouldn’t / that / the) tuition fees next year.</p>`

![gapfill - marked incorrect](https://user-images.githubusercontent.com/119390375/204438643-7e80201a-30b3-4f7a-8f12-8a96cd7aff2a.png)
![gap-fill - marked correct](https://user-images.githubusercontent.com/119390375/204438654-3f2ad3f8-29fc-46ca-985a-c3c60468794f.png)

Moodle Version: 3.9.3 (Build: 20201109)
Gapfill version: 2.134 (2022061900)

**Comment by marcusgreen** [comment]
could you provide an xml dump of your questions. From a very quick glance they look very straightforward but it is helpful if I am looking at all of  exactly the same as you. Thank you for providing the Moodle version and Gapfill version and text, it makes things much easier. The first thing I will look at is the comma in couldn't and how it is handled in code.  Also to check if it is actually a comma or possibly a backtick ` (or similar)...

**Comment by marcusgreen** [comment]
https://vledevelop.co.uk/gapfill/mod/quiz/attempt.php?attempt=25&cmid=3
username
s1
password
Password1!
CommaTest Quiz

**Comment by marcusgreen** [comment]
I am assuming the response is entered "by hand" rather than done via drag and drop, which makes me wonder if there is inconsistency in something, for example the spaces between words (which is not normalised). Could you copy and paste responses as text to look at that. I have thought about "normalising white space" in the past, but it is an edge condition.  If you are able to access that example quiz I just posted the link to, although it is set up for drag and drop you can still test it "by hand", for example you can copy and paste your student answers in to test it (though of course you can do that with your own setup). I can give you teacher access to a similar course if that would be helpful.

**Comment by conradteper** [comment]
Can confirm these are entered by hand - not drag drop or gap fill.
I've also checked for extra spaces as this is a common error for many of my students.

I've used the example quiz and copy pasted the following students' answers:
agree that the college shouldn’t raise
This has been graded as incorrect

agree that the college shouldn't raise
This has been graded as correct

I investigated your backtick/apostrophe theory
I edited the question as follows:
`<p>28. Many students [agree that the college shouldn’t raise|agree that the college shouldn't raise] (agree / college / raise / shouldn’t / that / the) tuition fees next year.</p>`

This has resolved the issue.

Thanks!
